lambda/MAS/Services/MailService.cs

```diff
@@ -113,6 +117,57 @@ public MailService(IMailChimpManager mailChimpManager, ILogger<MailService> logg
             {
                 _logger.LogError(e, $"Failed to send daily email for {dateStr}: {e.Message}");
                 throw new Exception($"Failed to send daily email for {dateStr}: {e.Message}", e);
+            }  
+        }
+          
+        public async Task<string> CreateAndSendWeeklyAsync(string subject, string previewText, string body, string date)
+        {
+            try
+            { 
+                var campaign = await _mailChimpManager.Campaigns.AddAsync(new Campaign
+                {
+                    Type = CampaignType.Regular,
+                    Settings = new Setting
+                    {
+                        FolderId = _mailChimpConfig.CampaignFolderId,
+                        TemplateId = _mailChimpConfig.WeeklyTemplateId,
+                        SubjectLine = string.Format(_mailConfig.WeeklySubject, date),
+                        FromName = _mailConfig.FromName,
+                        ReplyTo = _mailConfig.ReplyTo,
+                        PreviewText = previewText,
+                        AutoFooter = false
+                    },
+                    Recipients = new Recipient
+                    {
+                        ListId = _mailChimpConfig.ListId,
+                        SegmentOptions = new SegmentOptions
+                        {
+                            SavedSegmentId = _mailChimpConfig.WeeklySegmentId,
+                        }
+                    }
+                });
+                _logger.LogInformation($"Weekly campaign has been created with campaignId {campaign.Id}");
+                await _mailChimpManager.Content.AddOrUpdateAsync(campaign.Id, new ContentRequest
+                {
+                    Template = new ContentTemplate
+                    {
+                        Id = _mailChimpConfig.WeeklyTemplateId,
+                        Sections = new Dictionary<string, object> {
+                            { "body", body },
+                            { "date", date }
+                        }
+                    }
+                });
+                _logger.LogInformation($"The body of weekly template {_mailChimpConfig.DailyTemplateId} has been updated");
```

Comment:
This should be WeeklyTemplateId not daily, and it's body and date, not just body.